server/src/testFixtures/java/io/crate/testing/SQLTransportExecutor.java

```diff
@@ -396,7 +396,16 @@ public static Object toJdbcCompatObject(Connection connection, Object arg) {
             return DataTypes.STRING.implicitCast(arg);
         }
         if (arg.getClass().isArray()) {
```

Comment:
I think

```java
        if (arg instanceof Object[] values) {
            arg = Arrays.asList(values);
        }
```

Might be enough. I think it does the right thing for `float[]` without any translation.

Reply:
Yes but it avoids the class cast error and I think the jdbc client can deal with float[] directly, without further conversation